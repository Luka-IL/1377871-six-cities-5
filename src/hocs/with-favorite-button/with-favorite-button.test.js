import React from "react";
import renderer from "react-test-renderer";
import withFavoriteButton from "./with-favorite-button";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const store = mockStore({
  DATA: {
    active:
      {
        title: `1Beautiful & luxurious studio at great location`,
        premum: false,
        type: `apartment`,
        price: 20,
        rating: 4
      },
  },
  USER: {
    authorizationStatus: `AUTH`
  }
});
const MockComponent = () => {
  return (
    <React.Fragment />
  );
};

const MockComponentWrapped = withFavoriteButton(MockComponent);

it(`WithReviewsForm is rendered correctly`, () => {
  const tree = renderer.create((
    <Provider store={store}>
      <MockComponentWrapped
        loadOffersList={() => {}}
        changeFavoriteState={() => {}}
        loadFavoritesList={() => {}}
        id={0}
        authorizationStatus={`AUTH`}
        redirectToRoute={() => {}}
        favorite={true}
      />
    </Provider>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
