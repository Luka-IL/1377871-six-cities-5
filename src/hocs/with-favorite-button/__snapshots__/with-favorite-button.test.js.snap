// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithReviewsForm is rendered correctly 1`] = `null`;
